benchmark/Streamly/Benchmark/Data/Mut/Array.hs

```diff
@@ -0,0 +1,96 @@
+-- |
+-- Module      : Streamly.Benchmark.Data.Mut.Array
+-- Copyright   : (c) 2020 Composewell Technologies
+--
+-- License     : BSD-3-Clause
+-- Maintainer  : streamly@composewell.com
+
+{-# LANGUAGE CPP #-}
+{-# LANGUAGE FlexibleContexts #-}
+{-# LANGUAGE ScopedTypeVariables #-}
+
+#ifdef __HADDOCK_VERSION__
+#undef INSPECTION
+#endif
+
+#ifdef INSPECTION
+{-# LANGUAGE TemplateHaskell #-}
+{-# OPTIONS_GHC -fplugin Test.Inspection.Plugin #-}
+#endif
+
+module Main
+  (
+    main
+  ) where
+
+import Control.DeepSeq (NFData(..))
+import System.Random (randomRIO)
+import Prelude hiding ()
+
+import qualified Streamly.Prelude  as Stream
+import qualified Streamly.Internal.Data.Fold as Fold
+import qualified Streamly.Internal.Data.Array.Stream.Mut.Foreign as MArray
```

Comment:
We should import the array APIs from `Data.Array.Foreign.Mut`.

Reply:
arraysOf :: (IsStream t, MonadIO m, Storable a)
    => Int -> t m a -> t m (Array a) is not defined here (Data.Array.Foreign.Mut). Do we need to redefine here?

Reply:
We should not be using arraysOf, see my other comment. And even if we need that we can import that from this module and import other stuff from mutable array module.